File: trees/cppRealizations/sameTree.cpp
#include "btf.h"


class Solution {
public:
    bool isSameTree(TreeNode* p, TreeNode* q) {
        if (p == nullptr && q == nullptr) {
            return true;
        }
        if (p == nullptr || q == nullptr) {
            return false;
        }
        return p->val == q->val && isSameTree(p->left, q->left) && isSameTree(p->right, q->right);
    }
};


int main(){
    TreeNode* tree1 = new TreeNode(1);
    tree1->left = new TreeNode(2);
    tree1->right = new TreeNode(3);
    TreeNode* tree2 = new TreeNode(1);
    tree2->left = new TreeNode(2);
    tree2->right = new TreeNode(3);
    cout << "Input:" << endl;
    cout << "Tree1:" << endl;
    btf(tree1);
    cout << "Tree2:" << endl;
    btf(tree2);
    cout << "Solution:" << endl;
    cout << Solution().isSameTree(tree1, tree2) << endl;
    return 0;
}

File: trees/sameTree.py
from trees.btf import *


class Solution:
    def isSameTree(self, p: TreeNode, q: TreeNode) -> bool:
        if p is None and q is None:
            return True
        if p is None or q is None:
            return False
        return p.val == q.val and self.isSameTree(p.left, q.left) and self.isSameTree(p.right, q.right)


if __name__ == '__main__':
    root1 = TreeNode(1)
    root1.left = TreeNode(2)
    root1.right = TreeNode(3)

    root2 = TreeNode(1)
    root2.left = TreeNode(2)
    root2.right = TreeNode(3)

    print("Input:")
    print("Tree 1:")
    btf(root1)
    print("Tree 2:")
    btf(root2)
    print("Sulution:")
    print(Solution().isSameTree(root2, root2))